backend/overstroomik_service/geoserver.py

```diff
@@ -0,0 +1,152 @@
+"""
+This class uses the geoserver-api and geoserver layers to find
+location information by rd coordinates
+"""
+
+from typing import Optional
+import httpx
+
+from overstroomik_service.auto_models import Data
+from overstroomik_service.config import settings
+
+ERROR_GENERAL_NOER = "no-error"
+ERROR_GEOS_NO_RESP = "Geen response geoserver"
+ERROR_GEOS_NO_SMAP = "Geen response specifieke kaart"
+
+
+class Geoserver():
+
+    async def get_data(self,
+                       rd_x: float,
+                       rd_y: float,
+                       geoserver_url: Optional[str] = settings.GEOSERVER_URL,
+                       layers: [str] = settings.GEOSERVER_LAYER):
+        """
+        Find location information with specified rd coordinates.
+        :param rd_x: x coordinate in meters, geocoded location longitude transformed into EPSG:28992
+        :param rd_y: y coordinate in meters, geocoded location latitude transformed into EPSG:28992
+        :param geoserver_url: link to the geoserver (example: http://geoserver:8080/geoserver)
+        :param layers: group layer with the expected data  (example: overstroomik:Overstroomik_data)
+        """
+
+        # initial no error
+        status = ERROR_GENERAL_NOER
+
+        # start with empty object
+        data_item = {}
+
+        # create the getfeature info url
+        api_info = self.get_api_url_from_rd(
+            rd_x=rd_x, rd_y=rd_y, geoserver_url=geoserver_url)
+
+        url = api_info.get("api_url")
+
+        # Check input, is location between the layer exent
+        if api_info.get("valid", True) is False:
+            status = ERROR_GEOS_NO_SMAP
+        else:
+            # connect async to the geoserver
+            async with httpx.AsyncClient() as client:
+
+                # fetch the feature info
+                result = await client.get(url, timeout=10.0)
+
+                if result.status_code == httpx.codes.OK:
+
+                    out = result.json()
```

Comment:
From here on there's way to deep nesting. Move all the parsing logic into a different function.